README.md

```diff
@@ -0,0 +1,241 @@
+
+When referencing icons with the [Blade directive](#directives) or [helper](#helper) you can omit the prefix to reference icons from the `default` set.
+
+### Default Classes
+
+You can optionally define classes which will be applied to every icon by filling in the `class` setting in your `blade-icons.php` config file:
+
+```php
+<?php
+
+return [
+    'class' => 'icon icon-default',
+];
+```
+
+If you don't want any classes to be applied by default then leave this as an empty string.
+
+## Usage
+
+There are several ways of inserting icons into your Blade templates. We personally recommend using Blade components, but you can also make use of Blade directives if you wish.
+
+### Components
+
+The easiest way to get started with using icons from sets are Blade components:
+
+```blade
+<x-icon-camera/>
+```
+
+Icons in subdirectories can be referenced using dot notation:
+
+```blade
+<x-icon-solid.camera/>
+```
+
+You can also pass classes to your icon components (default classes will be applied as well):
+
+```blade
+<x-icon-camera class="icon-lg"/>
+```
+
+Or any other attributes for that matter:
+
+```blade
+<x-icon-camera class="icon-lg" id="settings-icon" style="color: #555" data-baz/>
+```
+
+> Note that with Blade components, using a prefix is always required, even when referencing icons from the default set.
+
+### Directives
+
+If components aren't really your thing you can make use of the Blade directive instead. If you defined a default `icon` class in your config and want to render a `camera` icon with an `icon-lg` class you can do that like so:
+
+```blade
+@svg('camera', 'icon-lg')
+```
+
+Any additionally attributes can be passed as a third array argument, and they'll be rendered on the `svg` element:
+
+```blade
+@svg('camera', 'icon-lg', ['id' => 'settings-icon'])
+```
+
+If you don't have a class to be defined you can also pass these attributes as the second parameter:
+
+```blade
+@svg('camera', ['id' => 'settings-icon'])
+```
+
+If you want to override the default classes, pass in the class as an attribute:
+
+```blade
+@svg('camera', ['class' => 'icon-lg'])
+```
+
+Attributes without a key, are supported too:
+
+```blade
+@svg('camera', ['data-foo'])
+```
+
+### Helper
+
+If you'd like, you can use the `svg` helper to expose a fluent syntax for setting SVG attributes:
+
+```blade
+{{ svg('camera')->id('settings-icon')->dataFoo('bar')->dataBaz() }}
+```
+
+## Building Packages
+
+If you're interested in building your own third party package to integrate an icon set, it's pretty easy to do so. Make sure to load your SVGs from the `boot` method of your package's service provider. Provide the path to your SVGs and provide your own unique set name and component prefix:
+
+```php
+use BladeUI\Icons\Factory;
+
+public function boot(): void
+{
+    $this->app->make(Factory::class)->add('heroicons', [
+        'path' => __DIR__ . '/../resources/svg',
+        'component-prefix' => 'heroicon',
```

Comment:
@driesvints I was trying to follow the package creation and I guess it should be `prefix` and not `component-prefix` the configuration key to use

```suggestion
        'prefix' => 'heroicon',
```

Reply:
Thanks! Fixed.